cirq-core/cirq/sim/simulator.py

```diff
@@ -814,19 +813,18 @@ def __init__(
         self,
         params: study.ParamResolver,
         measurements: Dict[str, np.ndarray],
-        final_simulator_state: Any,
+        final_step_result: StepResult,
```

Comment:
This API change is backwards-incompatible, as subclasses of this type (outside of Cirq) assume that this will be `final_simulator_state`. Additionally, simulators which only implement `SimulatesFinalState` cannot use this, as `SimulatesFinalState` does not generate `StepResult`s.

Instead, we should leave the API as it was and only separate `step_result` vs `simulator_state` in the member variables of the class.

Example: [TrialResult from qsim](https://github.com/quantumlib/qsim/blob/d1a4bfa1c3aeef86a67c93e0b7fc849ef3070bc8/qsimcirq/qsim_simulator.py#L55)

Reply:
In contrast, the `_create_simulator_trial_result` API change is fine, since it is private.

Reply:
Done